### PR TITLE
Better kick targets

### PR DESCRIPTION
**Pull request by h3ndrk**
## Introduced Changes

Rebase improved kick targets if the ball is in the opponents corners.

Based on #268 

## ToDo / Known Issues

None

## Ideas for Next Iterations (Not This PR)

None

## How to Test

Deploy, let the robot walk into opponent half, place ball in the opponents corner and observe a single kick target at the penalty spot in Twix.

### CODE REVIEW

File: crates/control/src/behavior/dribble.rs [hunk 1]
```diff
@@ -249,6 +245,29 @@ fn find_targets_to_kick_to(
         possible_kick_targets.extend(emergency_targets);
     }
 
+    // Create obstacles kick targets to filter with
```
Comment (schmidma) [inline crates/control/src/behavior/dribble.rs:248]: Move all these structural comments to function names?

File: crates/control/src/behavior/dribble.rs [hunk 2]
```diff
@@ -345,3 +364,20 @@ fn ball_is_close_to_own_goal(
...
+) -> bool {
+    let global_ball = robot_to_field * ball_position;
+    let left_opponent_corner = point![field_dimensions.length / 2.0, field_dimensions.width / 2.0];
+    let right_opponent_corner =
+        point![field_dimensions.length / 2.0, -field_dimensions.width / 2.0];
+    let ball_near_left_opponent_corner = distance(&global_ball, &left_opponent_corner)
+        < parameters.distance_from_corner;
+    let ball_near_right_opponent_corner = distance(&global_ball, &right_opponent_corner)
+        < parameters.distance_from_corner;
+    ball_near_left_opponent_corner || ball_near_right_opponent_corner
```
Comment (schmidma) [inline crates/control/src/behavior/dribble.rs:382]: some `.any()`?

**Review by schmidma — CHANGES_REQUESTED** [review]

File: crates/control/src/behavior/dribble.rs (hunk 2 above)
Comment (h3ndrk) [reply crates/control/src/behavior/dribble.rs:382]: Could you elaborate? 🤔 These are two booleans. Do you suggest to use an array to be able to call `.any()` on it?

Comment (h3ndrk) [reply crates/control/src/behavior/dribble.rs:394]: IMHO, it is clear enough without arrays and would only get more complex. It also fits the style of the rest of the file.

File: crates/control/src/behavior/dribble.rs [hunk 3]
```diff
@@ -190,48 +192,42 @@ fn find_targets_to_kick_to(
...
-            };
-            Circle {
-                center,
-                radius: obstacle_radius,
-            }
-        })
-        .collect();
 
-    let mut possible_kick_targets = vec![left_goal_half, right_goal_half];
+    let mut possible_kick_targets: Vec<Point2<f32>> = vec![];
```
[the remark this answers is not in the document]
Comment (h3ndrk) [reply crates/control/src/behavior/dribble.rs]: Jep, also removed the type annotation.

File: crates/control/src/behavior/dribble.rs [hunk 4]
```diff
@@ -190,48 +192,42 @@ fn find_targets_to_kick_to(
...
+
+    // Create from corner targets
+    if is_ball_in_opponents_corners(&ball_position, parameters, field_dimensions, robot_to_field)
+        && parameters.use_corner_kick_targets
```
[the remark this answers is not in the document]
Comment (h3ndrk) [reply crates/control/src/behavior/dribble.rs]: Removed.

File: etc/configuration/default.json [hunk 5]
```diff
@@ -922,13 +922,18 @@
           "enabled": true
         }
       },
+      "find_kick_targets": {
+        "use_corner_kick_targets": false,
```
[the remark this answers is not in the document]
Comment (h3ndrk) [reply etc/configuration/default.json]: Removed.

File: crates/control/src/behavior/dribble.rs (hunk 1 above)
Comment (h3ndrk) [reply crates/control/src/behavior/dribble.rs]: Done

File: crates/control/src/kick_selector.rs [hunk 6]
```diff
@@ -247,19 +249,23 @@ fn collect_kick_targets(
...
-            -field_dimensions.goal_inner_width / 4.0
-        ];
-    let kick_targets = vec![left_goal_half, right_goal_half];
+    let mut kick_targets = Vec::new();
+
+    if is_ball_in_opponents_corners(&ball_position, parameters, field_dimensions, robot_to_field) {
+        kick_targets.extend(generate_corner_kick_targets(
+            parameters,
+            field_dimensions,
+            robot_to_field,
```
Comment (schmidma) [inline crates/control/src/kick_selector.rs:261]: use `field_to_robot` here

Comment (h3ndrk) [reply crates/control/src/kick_selector.rs:261]: Done

**Review by schmidma — APPROVED** [review]

**Pull request merged**